app/src/main/java/me/timpushkin/sgbmandroidapp/MainActivity.kt

```diff
@@ -0,0 +1,155 @@
+package me.timpushkin.sgbmandroidapp
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.net.Uri
+import android.os.Bundle
+import android.os.SystemClock
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.asImageBitmap
+import kotlinx.coroutines.launch
+import me.timpushkin.sgbmandroid.DepthEstimator
+import me.timpushkin.sgbmandroidapp.ui.MenuButtons
+import me.timpushkin.sgbmandroidapp.ui.ScaleFactorField
+import me.timpushkin.sgbmandroidapp.utils.StorageUtils
+import me.timpushkin.sgbmandroidapp.utils.depthArrayToBitmap
+
+private const val CACHED_PARAMS = "params.xml"
+private const val CACHED_LEFT_IMAGE = "left.png"
+private const val CACHED_RIGHT_IMAGE = "right.png"
+private const val CACHED_DEPTH_MAP = "depth.png"
+
+class MainActivity : ComponentActivity() {
+    private lateinit var mStorageUtils: StorageUtils
+    private lateinit var mDepthEstimator: DepthEstimator
+
+    private var mImages: Pair<ByteArray, ByteArray>? = null
+    private var mDepthMap by mutableStateOf<Bitmap?>(null)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        mStorageUtils = StorageUtils(this)
+        with(mStorageUtils) {
+            getFromCache(CACHED_PARAMS)?.run { mDepthEstimator = DepthEstimator(path) }
```

Comment:
I think here it is too much for lambdas. Yes, they are great, but here only person with strong kotlin knowledge can understand where `path` is taken))

Reply:
I decided to use `ViewModel` instead of caching, because there is no actual reason to persist image pair between app invocations, I believe, so now this line is gone.